pkg/check/node_uuid.go

```diff
@@ -0,0 +1,21 @@
+package check
+
+import (
+	"fmt"
+
+	"github.com/vmware/govmomi/vim25/mo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// CheckNodeDiskUUID makes sure that all nodes have disk.enableUUID=TRUE.
+func CheckNodeDiskUUID(ctx *CheckContext, node *v1.Node, vm *mo.VirtualMachine) error {
+	if vm.Config.Flags.DiskUuidEnabled == nil {
+		return fmt.Errorf("the node has empty disk.enableUUID")
+	}
+	if *vm.Config.Flags.DiskUuidEnabled == false {
+		return fmt.Errorf("the node has disk.enableUUID = FALSE")
+	}
+	klog.V(4).Infof("... the node has correct disk.enableUUID")
+	return nil
+}
```

Comment:
Not related to this PR but I am not convinced that we should emit duration it takes to perform these checks as metrics (histogram). Specially those that are at node level. This causes explosion of time series that must be persisted for 1 year or so. Do we really need customers to pay additional storage/cpu for storing the metrics that they may never look at.

Reply:
I am open for suggestions (in a separate PR :-).